### ISSUE TITLE
Do not append, instead clobber.

### ISSUE THREAD

**Issue by jordwalke** [MikkoH/npm-hooks#1]
I found it much more useful in my fork of npm-hooks, to clobber instead of append the file. Do you agree this should be a supported mode?

**Comment by mikkoh** [comment]
Yep you're probably right. Could allow clobbering via option. Not sure what the default should be.

I haven't used this module in awhile. Initially I was wanting to create a module that would track installed modules. A sort of analytics module.

**Comment by jordwalke** [comment]
I found an incredibly important use case for this (or hooks in general). We created a far superior postinstall for the purpose of architecture specific compilation, but it could be used for any build step in postinstall. It's called `esy`.

- Parallel builds among all packages.
- Caches artifacts across projects (important for native compilation which takes minutes/hours).
- Recoverable builds.
- Out of source builds - doesn't touch your actual node_modules and write artifacts into them.

Here's what the package.json would look like:

```json
{
  "name": "testtop",
  "version": "1.0.0",
  "scripts": {
    "install": "esybuild"
  },
  "esy": {
    "build": ["your commands here"]
  },
  "dependencies": {
    "esy": "*"
  }
}
```


[Here's](https://github.com/reasonml/esy/commit/787715cd73e3e1acff248f5a4df95f8fb853dac3) where I use my fork of this library to do that. What it does is, it attaches a hook to ensure that only the top level package's post install runs, which kicks of the parallel builds from the top down in a system we control, not `npm`.

Notice how it just *looks* like you're running `esybuild` on `install` - which you are, but that's not the real build :D
